lib/channel-postmessage/src/index.ts

```diff
@@ -231,8 +231,8 @@ const getEventSourceUrl = (event: MessageEvent) => {
 
   if (frame && remainder.length === 0) {
     const src = frame.getAttribute('src');
-    const { origin, pathname } = new URL(src, document.location);
-    return origin + pathname;
+    const { protocol, host, pathname } = new URL(src, document.location);
+    return `${protocol}//${host}${pathname}`;
```

Comment:
I think this will drop the port, if there's 1 present.

Reply:
According to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/URL/host), It seems that `host` includes the port.

Reply:
I stand corrected 🙇